main/accounts/Account/index.ts

```diff
@@ -236,73 +237,99 @@ class FrameAccount {
+
+    const contract = new Erc20Contract(contractAddress, req.data.chainId)
     const decodedData = contract.decodeCallData(calldata)
 
-    if (decodedData && Erc20Contract.isApproval(decodedData)) {
-      const spender = decodedData.args[0].toLowerCase()
-      const amount = decodedData.args[1].toHexString()
-      const { decimals, name, symbol } = await contract.getTokenData()
-
-      this.addRequiredApproval(
-        req,
-        new BigNumber(amount).isZero() ? ApprovalType.TokenSpendRevocation : ApprovalType.TokenSpendApproval,
-        {
-          decimals,
-          name,
-          symbol,
-          amount,
-          contract: contractAddress,
-          spender
-        },
-        (data: { amount: string }) => {
-          // amount is a hex string
-          const approvedAmount = new BigNumber(data.amount).toString()
-          log.info(`changing approved amount for request ${req.handlerId} to ${approvedAmount}`)
-
-          req.data.data = contract.encodeCallData('approve', [spender, data.amount])
-
-          if (req.decodedData) {
-            req.decodedData.args[1].value = approvedAmount
+    if (decodedData) {
+      if (Erc20Contract.isApproval(decodedData)) {
+        const spender = decodedData.args[0].toLowerCase()
+        const amount = decodedData.args[1].toHexString()
+        const { decimals, name, symbol } = await contract.getTokenData()
+  
+        this.addRequiredApproval(
+          req,
+          new BigNumber(amount).isZero() ? ApprovalType.TokenSpendRevocation : ApprovalType.TokenSpendApproval,
+          {
+            decimals,
+            name,
+            symbol,
+            amount,
+            contract: contractAddress,
+            spender
+          },
+          (data: { amount: string }) => {
+            // amount is a hex string
+            const approvedAmount = new BigNumber(data.amount).toString()
+            log.info(`changing approved amount for request ${req.handlerId} to ${approvedAmount}`)
+  
+            req.data.data = contract.encodeCallData('approve', [spender, data.amount])
+  
+            if (req.decodedData) {
+              req.decodedData.args[1].value = approvedAmount
+            }
           }
-        }
-      )
-
-      this.update()
-    }
-  }
-
-  private async populateRequestCallData (req: TransactionRequest, calldata: string) {
-    try {
-      const decodedData = await decodeContractCall(req.data.to || '', calldata)
-      const knownTxRequest = this.requests[req.handlerId] as TransactionRequest
-
-      if (knownTxRequest && decodedData) {
-        knownTxRequest.decodedData = decodedData
+        )
         this.update()
-      } 
-    } catch (e) {
-      log.warn(e)
+      }
     }
   }
 
-  private async populateRequestEnsName (req: TransactionRequest) {
-    const { to } = req.data
+  private async revealDetails (req: TransactionRequest) {
+    const tx = req.data
+    const calldata = tx.data
+    const chainId = tx.chainId
```

Comment:
probably up to preference but can one line these to `const { to, chainId, data: calldata } = req.data` and remove all the `tx.` parts to references below